### PR TITLE
Blacklisting of everything in the arsenal

### PR DESCRIPTION
**Pull request by Wyqer**
As it was only possible to blacklist backpack items in the arsenal via
classnames_extension.sqf I've added the ability to blacklist also
weapons, items, scopes, magazines, uniforms, explosives, etc. from the
arsenal.

### CODE REVIEW

**Comment by GreuhZbug** [comment]
Did you test it ? The reason blacklisting was limited to backpacks is because it would take a full ten seconds to open the arsenal, up to several minutes if you had enough mods. Maybe the BIS arsenal was updated since then to make it more efficient, but a few monthes ago that was definitely not the case.

The reason it's so slow is because the arsenal doesn't have a blacklist function. What the script does instead is to whitelist every single item in the game except those you've blacklisted. That's definitely not the best way to do it, but we don't really get a choice

**Comment by Wyqer** [comment]
Yeah, it's still quite a "workaround", but it works.
I've tested it on a local MP Session and as I also was curious about the loading time I can say it takes the first time about 2-3 Seconds. After the first time it get's immediately. On next wednesday we'll try this feature at our event evening on our liberation takistan map. If I encounter huge loading times in a dedicated server environment with about 12-15 players I'll let you know.

**Comment by Wyqer** [comment]
And about the "up to several minutes if you had enough mods".
I think you provide a great framework for coop events with this mission. And as I know many people would like to just blacklist the nightstalker or something like that without writing a whitelist with for example 999 from 1000 Items. And in my opinion, if a server admin would like to use for example 20 mods with each 100 new weapons and that would slow down the arsenal loading time it would be a problem of the server admin and not the framework developer.

**Comment by jus61** [comment]
I tested it on my server and everything is OK no loading times in Arsenal

Server : 178.32.49.181:2302

File: classnames_extension.sqf [hunk 1]
```diff
@@ -302,6 +302,26 @@ elite_vehicles_extension = [
...
+blacklisted_from_arsenal_weapons = [
+	"rhs_weap_M107",
+	"rhs_weap_M107_d",
+	"rhs_weap_M107_w"
+];
+
+// Classnames of items (optics, uniforms, vests, etc.) which should not be available in the arsenal
+blacklisted_from_arsenal_items = [
+	"optic_Nightstalker",
+	"optic_tws"
```
Comment (jus61) [inline classnames_extension.sqf:316]: "optic_tws_mg"

**Pull request closed without merging**